### ISSUE TITLE
PodGroup Pending forever because previous unschedulable podgroup occupied resources

### ISSUE THREAD

**Issue by Antsypc** [volcano-sh/volcano#2814]
## What happened
PodGroup is pending but accually there is enough idle resources.

## How to reproduce it (as minimally and precisely as possible)
- Cluster: 2 nodes with 1 GPU each.
- Firstly, submit a Job with one task using 2 GPUs. Of course, PodGroup of this Job is Inqueue, but Pod is pending.
- Secondly, submit a job using 1 GPU.
    - PodGroup is pending, pod will not be created. (PodGroup should be Inqueue, that's what I expected)
    - Obviously, 2 GPUs are idle, not occupied by any Pod. The second Job should be Running.

## Anything else we need to know
Before volcano 1.5.0, the `proportion` plugin always enqueue the job If capability of queue is not set.

```go
	ssn.AddJobEnqueueableFn(pp.Name(), func(obj interface{}) int {
		job := obj.(*api.JobInfo)
		queueID := job.Queue
		attr := pp.queueOpts[queueID]
		queue := ssn.Queues[queueID]
		// If no capability is set, always enqueue the job.
		if len(queue.Queue.Spec.Capability) == 0 {
			klog.V(4).Infof("Capability of queue <%s> was not set, allow job <%s/%s> to Inqueue.",
				queue.Name, job.Namespace, job.Name)
			return util.Permit
		}

		if job.PodGroup.Spec.MinResources == nil {
			klog.V(4).Infof("job %s MinResources is null.", job.Name)
			return util.Permit
		}
		minReq := job.GetMinResources()

		klog.V(5).Infof("job %s min resource <%s>", job.Name, minReq.String())
		klog.V(5).Infof("queue %s capability <%s>", queue.Name, api.NewResource(queue.Queue.Spec.Capability).String())
		klog.V(5).Infof("queue %s allocated <%s>", queue.Name, attr.allocated.String())
		// The queue resource quota limit has not reached
		inqueue := minReq.Add(attr.allocated).Add(attr.inqueue).LessEqual(api.NewResource(queue.Queue.Spec.Capability), api.Infinity)
		if inqueue {
			attr.inqueue.Add(job.GetMinResources())
			return util.Permit
		}
		return util.Reject
	})
```

However, since then, the condition to judge if capability of queue is not set has changed.
```go
	ssn.AddJobEnqueueableFn(pp.Name(), func(obj interface{}) int {
		job := obj.(*api.JobInfo)
		queueID := job.Queue
		attr := pp.queueOpts[queueID]
		queue := ssn.Queues[queueID]
		// If no capability is set, always enqueue the job.
		if attr.realCapability == nil {
			klog.V(4).Infof("Capability of queue <%s> was not set, allow job <%s/%s> to Inqueue.",
				queue.Name, job.Namespace, job.Name)
			return util.Permit
		}

		if job.PodGroup.Spec.MinResources == nil {
			klog.V(4).Infof("job %s MinResources is null.", job.Name)
			return util.Permit
		}
		minReq := job.GetMinResources()

		klog.V(5).Infof("job %s min resource <%s>, queue %s capability <%s> allocated <%s>",
			job.Name, minReq.String(), queue.Name, attr.realCapability.String(), attr.allocated.String())
		// The queue resource quota limit has not reached
		inqueue := minReq.Add(attr.allocated).Add(attr.inqueue).LessEqual(attr.realCapability, api.Infinity)
		klog.V(5).Infof("job %s inqueue %v", job.Name, inqueue)
		if inqueue {
			attr.inqueue.Add(job.GetMinResources())
			return util.Permit
		}
		return util.Reject
	})
```

## Proposal
Revert `if attr.realCapability == nil` to `if len(queue.Queue.Spec.Capability) == 0`

## Environment
- Volcano Version: 1.7.0 (It could be reproduced since 1.5.0)
- Kubernetes version (use `kubectl version`): 1.19 (It dosen't matter, I think)

**Comment by wangyang0616** [comment]
The reason for the `if attr.realCapability == nil ` modification is to support the `guarantee `function of the queue. If it is rolled back, the guarantee function will be abnormal.

**Comment by Antsypc** [comment]
@wangyang0616 
For now, I have to disable this `proportion` plugin. Is there a better way?

**Comment by wangyang0616** [comment]
In this week's regular meeting, the community discussed the current issue, and the main information is as follows.

**question:**
If the user does not set the capability of the queue, should we submit unlimited vcjobs according to v1.5, or is it more reasonable to judge realcapability according to v1.7?

**discuss:**
If you use `guarantee` related capabilities, you need to use the realcapability judgment. If the user does not pay attention to `guarantee`, it is more reasonable to use the capability set by the user.

**in conclusion:**
The introduction of `guarantee` is an allocation scenario related to elastic queues. For most user scenarios, it is not a rigidly needed function. We can add the following configuration to the default configuration of the trunk. The guarantee capability is disabled by default. If users need it, they can manually enable it.
```
-name: proportion
    enableJobEnqueued: false
```

**Comment by Antsypc** [comment]
@wangyang0616  Thanks for your help. Setting `enableJobEnqueued: false` works for me.

**Comment by xiao-jay** [comment]
@wangyang0616 
> The guarantee capability is disabled by default. If users need it, they can manually enable it.

Now looks like the guarantee capability is enabled by default,I am willing to contribute to this issue，/assign
<img width="1107" alt="image" src="https://github.com/volcano-sh/volcano/assets/87080562/71dae110-0bc6-4706-aa12-73af4b658c86">

**Comment by lowang-bh** [comment]
I think change `attr.realCapability == nil` to `attr.realCapability.IsEmpty()` will be compatible with both situation.
Becaulse `attr.realCapability ` looks will nevel be nil
```go
	realCapability := pp.totalResource.Clone().Sub(pp.totalGuarantee).Add(attr.guarantee)
	if attr.capability == nil {
		attr.realCapability = realCapability
	} else {
		attr.realCapability = helpers.Min(realCapability, attr.capability)
	}
```

**Comment by wangyang0616** [comment]
> I think change `attr.realCapability == nil` to `attr.realCapability.IsEmpty()` will be compatible with both situation. Becaulse `attr.realCapability ` looks will nevel be nil
> 
> ```go
> 	realCapability := pp.totalResource.Clone().Sub(pp.totalGuarantee).Add(attr.guarantee)
> 	if attr.capability == nil {
> 		attr.realCapability = realCapability
> 	} else {
> 		attr.realCapability = helpers.Min(realCapability, attr.capability)
> 	}
> ```

Judging from the context, `attr.realCapability == nil` is indeed always false, but changing to `attr.realCapability.IsEmpty()` should also be false, regardless of whether the user sets Capability, attr.realCapability will be assigned.
